### DISCUSSION TITLE
Securing Admin Token

### DISCUSSION THREAD

**Discussion by Gugulator** [hassio-addons/addon-bitwarden#280] [Q&A]
Looking for how to remove admin token nag message about plain text password stored in system.
![image](https://github.com/hassio-addons/addon-bitwarden/assets/93097630/cf59d838-b4cc-4032-a2b0-ad2f3c0be522)
I'm using HA OS as VM.

successfully ran this command:
`docker exec -it addon_a0d7b954_bitwarden /opt/vaultwarden hash`

and got this result:
`Generate an Argon2id PHC string using the 'bitwarden' preset:
Password:
Confirm Password:
ADMIN_TOKEN='$argon2id$v=19$m=65540,t=3,p=very_long_value_blah-blah-blah'`

I also found that plain text admin token possible to see via this command:
`docker exec -it addon_a0d7b954_bitwarden cat /data/config.json`

To secure using admin token, should I edit config.json file and if yes, what exactly need to be written there?

**Comment by Stephan-4711** [comment]
Did you find a solution?
I'm not able to change the admin token.
When I change the token in the admin interface, the logon is broken afterwards.

**Comment by Slin-nilS** [comment]
> docker exec -it addon_a0d7b954_bitwarden cat /data/config.json

Thank you for your Information about to see the token. That helped me.

But now to your question:
You can simply insert the new token in the admin Interface. 
> When I change the token in the admin interface, the logon is broken afterwards.
I have no logon issues afterwards. Wich command you use for the generate? I used:
```ssh
docker exec -it addon_YOURCONTAINERID_bitwarden /opt/vaultwarden hash
```
I hope I was able to help.
